broadcast/src/linked/prover.rs

```diff
@@ -0,0 +1,103 @@
+use super::{namespace, serializer, wire, Context, Epoch};
+use crate::Proof;
+use bytes::{Buf, BufMut};
+use commonware_cryptography::{
+    bls12381::primitives::{
+        group::{self, Element},
+        ops,
+    },
+    Digest, Scheme,
+};
+use std::marker::PhantomData;
+
+#[derive(Clone)]
+pub struct Prover<C: Scheme, D: Digest> {
+    _crypto: PhantomData<C>,
+    _digest: PhantomData<D>,
+
+    public: group::Public,
+    namespace: Vec<u8>,
+}
+
+impl<C: Scheme, D: Digest> Prover<C, D> {
+    /// Create a new prover with the given signing `namespace`.
+    pub fn new(public: group::Public, namespace: &[u8]) -> Self {
+        Self {
+            _crypto: PhantomData,
+            _digest: PhantomData,
+            public,
+            namespace: namespace::ack(namespace),
+        }
+    }
+
+    /// Returns 1) the length of a proof and 2) a tuple with the lengths of:
+    /// - the digest
+    /// - the public key
+    /// - the signature
+    fn get_len() -> (usize, (usize, usize)) {
```

Comment:
Realizing now we might want to make length public to make encoding easier. We can do this after merging though.